### PR TITLE
Disable asserts, avoid line numbers in runtime checks.

### PR DESCRIPTION
**Pull request by olonho**

### CODE REVIEW

File: backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/llvm/IrToBitcode.kt [hunk 1]
```diff
@@ -2626,13 +2627,20 @@ internal class CodeGeneratorVisitor(val context: Context, val lifetimes: Map<IrE
...
-            LLVMSetLinkage(llvmUsedGlobal.llvmGlobal, LLVMLinkage.LLVMAppendingLinkage)
-            LLVMSetSection(llvmUsedGlobal.llvmGlobal, "llvm.metadata")
+        LLVMSetLinkage(llvmUsedGlobal.llvmGlobal, LLVMLinkage.LLVMAppendingLinkage)
+        LLVMSetSection(llvmUsedGlobal.llvmGlobal, "llvm.metadata")
+    }
+
+    private fun appendDebugSelector() {
+        if (context.config.produce.isNativeBinary) {
+            val llvmDebugSelector =
+                    context.llvm.staticData.placeGlobal("KonanNeedDebugInfo",
```
Comment (SvyatoslavScherbina) [inline backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/llvm/IrToBitcode.kt]: Marking this global as constant may also have effect to optimizer.

File: backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/llvm/IrToBitcode.kt [hunk 2]
```diff
@@ -2626,13 +2627,20 @@ internal class CodeGeneratorVisitor(val context: Context, val lifetimes: Map<IrE
...
+                            Int32(if (context.shouldContainDebugInfo()) 1 else 0))
+            LLVMSetLinkage(llvmDebugSelector.llvmGlobal, LLVMLinkage.LLVMExternalLinkage)
```
Comment (SvyatoslavScherbina) [inline backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/llvm/IrToBitcode.kt]: Why not pass `isExternal = true` to `placeGlobal` instead?

File: backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/llvm/IrToBitcode.kt (hunk 1 above)
Comment (olonho) [reply backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/llvm/IrToBitcode.kt]: OK

File: backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/llvm/IrToBitcode.kt (hunk 2 above)
Comment (olonho) [reply backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/llvm/IrToBitcode.kt]: It only got `isExported` it seems.

Comment (SvyatoslavScherbina) [reply backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/llvm/IrToBitcode.kt]: Yes, my bad, I have meant `isExported`.

**Pull request merged**